src/transformers/models/big_bird/modeling_big_bird.py

```diff
@@ -1378,6 +1379,13 @@ def forward(
         from_blocked_mask=None,
         to_blocked_mask=None,
     ):
+        # fp16 compatibility
+        def convert_dtype(mask):
+            return tree_map(lambda x: x.to(hidden_states.dtype) if isinstance(x, torch.Tensor) else x, mask)
+
+        band_mask = convert_dtype(band_mask)
```

Comment:
Hey @xuzhao9 :-) 

Could we maybe just do:

```suggestion
if band_mask is not None:
        band_mask = band_mask.to(hidden_states.dtype)
```

This would be much more readable IMO :-) .

Reply:
same for the other two tensors

Reply:
That makes sense. I have addressed your comments.